relayer/relayer.go

```diff
@@ -387,3 +460,29 @@ func (r *Relayer) CheckSupportedDestination(destinationBlockchainID ids.ID) bool
 	supportedDsts := r.sourceBlockchain.GetSupportedDestinations()
 	return supportedDsts.Contains(destinationBlockchainID)
 }
+
+func (r *Relayer) CheckAllowedOriginSenderAddress(addr common.Address) bool {
+	allowedAddresses := r.globalConfig.GetSourceBlockchainAllowedAddresses(r.sourceBlockchain.GetBlockchainID())
+	if bytes.Equal(allowedAddresses[0].Bytes(), common.Address{}.Bytes()) {
```

Comment:
What's the rationale behind putting the empty address in the slice, instead of just checking for an empty slice?

Reply:
Really just to encapsulate the "all allowed" relayer key case in the `Config`, especially w.r.t. [fetching the relayer keys](https://github.com/ava-labs/awm-relayer/blob/5982596abfbd082441be59418089d9ae676d154a/database/database.go#L87).

The approach I went with requires the caller to understand that the zero address represents a wildcard, whereas keeping the slice empty would require the caller to understand that an empty slice represents a wildcard.

Reply:
I'm not a huge fan of the sentinel value, but not too picky. I think having the behaviour line up with what we expect in the config (empty list in the config means all allowed). As-is we could also panic if `GetSourceBlockchainAllowedAddresses` returns an empty slice. At the very least we should clearly document in/above each of the functions the expected behaviour.

Reply:
If we're worried about accidentally not having initialized the slice, we could check for a empty non-nil slice?

Reply:
I went ahead and implemented your suggestion, and agree it's cleaner. It makes sense to have the relayer ID constructor interpret the empty list, rather than modifying the config object directly.